native/core/src/execution/shuffle/shuffle_writer.rs

```diff
@@ -625,22 +626,24 @@ impl MultiPartitionShuffleRepartitioner {
             return Ok(());
         }
 
-        let num_output_partitions = self.partition_writers.len();
-        let mut partitioned_batches = self.partitioned_batches();
-        let mut spilled_bytes = 0;
+        with_trace("shuffle_spill", self.tracing_enabled, || {
```

Comment:
```suggestion
        with_trace_async("shuffle_spill", self.tracing_enabled, || {
```

I just assumed if method is async then the `trace_async` needs to be called like in other similar methods?